### ISSUE TITLE
Add Tanh activation translator (sklearn MLP activation='tanh' and torch.nn.Tanh)

### ISSUE THREAD

**Issue by amol-** [posit-dev/orbital#116]
# Goal

Add a `Tanh` translator. Tanh is one of the four activations offered by sklearn's MLP (`activation="tanh"`) and a standard PyTorch layer (`torch.nn.Tanh`), but translation currently fails with `Translation for Tanh not implemented`. The implementation can follow the existing `Sigmoid` translator pattern.

Done criteria: both pipelines below translate and produce predictions matching the framework output (verified on DuckDB).

Real-world example (house price regression, tanh is a common pick for small regression MLPs — currently fails):

```python
import numpy as np
import pandas as pd
import orbital
from orbital.types import FloatColumnType
from sklearn.neural_network import MLPRegressor
from sklearn.pipeline import Pipeline
from sklearn.preprocessing import StandardScaler

rng = np.random.default_rng(7)
X = pd.DataFrame({
    "sqft": rng.uniform(500, 4_000, 800),
    "bedrooms": rng.integers(1, 6, 800).astype(float),
    "age_years": rng.uniform(0, 80, 800),
})
y = X["sqft"] * 210 + X["bedrooms"] * 9_000 - X["age_years"] * 400

pipeline = Pipeline([
    ("scaler", StandardScaler()),
    ("mlp", MLPRegressor(hidden_layer_sizes=(16,), activation="tanh",
                         max_iter=800, random_state=7)),
])
pipeline.fit(X, y)

orbital_pipeline = orbital.parse_pipeline(
    pipeline, features={c: FloatColumnType() for c in X.columns}
)
# raises: Translation for Tanh not implemented
```

Equivalent PyTorch model that should also work:

```python
import torch

model = torch.nn.Sequential(
    torch.nn.Linear(3, 16),
    torch.nn.Tanh(),
    torch.nn.Linear(16, 1),
)
orbital_pipeline = orbital.parse_pytorch_model(
    model, features={c: FloatColumnType() for c in ["sqft", "bedrooms", "age_years"]}
)
# raises: Translation for Tanh not implemented
```

# Why

Tanh is likely the first missing activation a real user hits: it is a first-class sklearn MLP option and common in small tabular networks. SQL backends expose `tanh()` natively, so the translator is a small addition with outsized coverage gains.

# References

- src/orbital/translation/steps/nn/sigmoid.py (SigmoidTranslator — the pattern to copy)
- src/orbital/translation/steps/nn/relu.py (another minimal activation translator)
- src/orbital/translate.py (TRANSLATORS map where the new op is registered)
